langchain4j-alloydb-pg/src/main/java/dev/langchain4j/store/embedding/alloydb/AlloyDBEmbeddingStore.java

```diff
@@ -37,52 +63,142 @@ public class AlloyDBEmbeddingStore implements EmbeddingStore<TextSegment> {
+            }
+        }
+
+        String query = "SELECT column_name, data_type FROM information_schema.columns WHERE table_name = \"" + tableName + "\" AND table_schema = \"" + schemaName + "\"";
+
+        Map<String, String> allColumns = new HashMap();
+
+        try (ResultSet resultSet = engine.getConnection().createStatement().executeQuery(query)) {
+            ResultSetMetaData rsMeta = resultSet.getMetaData();
+            int columnCount = rsMeta.getColumnCount();
+            for (int i = 1; i <= columnCount; i++) {
+                allColumns.put(rsMeta.getColumnName(i), rsMeta.getColumnTypeName(i));
+            }
+
+            if (!allColumns.containsKey(idColumn)) {
+                throw new IllegalStateException("Id column, " + idColumn + ", does not exist.");
+            }
+            if (!allColumns.containsKey(contentColumn)) {
+                throw new IllegalStateException("Content column, " + contentColumn + ", does not exist.");
+            }
+            if (!allColumns.get(contentColumn).equalsIgnoreCase("text") || !allColumns.get(contentColumn).contains("char")) {
+                throw new IllegalStateException("Content column, is type " + allColumns.get(contentColumn) + ". It must be a type of character string.");
+            }
+            if (!allColumns.containsKey(embeddingColumn)) {
+                throw new IllegalStateException("Embedding column, " + embeddingColumn + ", does not exist.");
+            }
+            if (!allColumns.get(embeddingColumn).equalsIgnoreCase("USER-DEFINED")) {
+                throw new IllegalStateException("Embedding column, " + embeddingColumn + ", is not type Vector.");
+            }
+            if (!allColumns.containsKey(metadataJsonColumn)) {
+                metadataJsonColumn = null;
+            }
+
+            for (MetadataColumn metadataColumn : metadataColumns) {
+                if (!allColumns.containsKey(metadataColumn.getName())) {
+                    throw new IllegalStateException("Metadata column, " + metadataColumn.getName() + ", is not type Vector.");
+                }
+            }
+
+            if (ignoredColumns != null && !ignoredColumns.isEmpty()) {
+
+                Map<String, String> allColumnsCopy = allColumns.entrySet().stream()
+                        .collect(Collectors.toMap(e -> e.getKey(), e -> e.getValue()));
+                ignoredColumns.add(idColumn);
+                ignoredColumns.add(contentColumn);
+                ignoredColumns.add(embeddingColumn);
+
+                for (String ignore : ignoredColumns) {
+                    allColumnsCopy.remove(ignore);
+                }
+
+                metadataColumns.addAll((allColumnsCopy.entrySet().stream()
+                        .map(e -> new MetadataColumn(e.getKey(), e.getValue(), true)).collect(Collectors.toList())));
+            }
+
+        } catch (SQLException ex) {
```

Comment:
Do not ignore errors, throw an error like throw new RuntimeException(String.format("Failed to execute '%s'", query), e);